From 3c12355ece0fa14e534c71a9f7327eda9773fad9 Mon Sep 17 00:00:00 2001
From: Stefan Schulze Frielinghaus <stefansf@linux.ibm.com>
Date: Mon, 9 Dec 2019 15:41:57 +0100
Subject: [PATCH] Fix endian handling w.r.t. CPP macro WORDS_BIGENDIAN

Include header file `ghcautoconf.h` where the CPP macro
`WORDS_BIGENDIAN` is defined. This finally fixes #17337 (in conjunction
with commit 6c59cc71dc).
---
 libraries/base/GHC/ByteOrder.hs                | 3 +++
 libraries/base/GHC/IO/Encoding/CodePage/API.hs | 3 +++
 testsuite/tests/codeGen/should_run/T7600.hs    | 4 ++++
 3 files changed, 10 insertions(+)

Index: b/libraries/base/GHC/ByteOrder.hs
===================================================================
--- a/libraries/base/GHC/ByteOrder.hs
+++ b/libraries/base/GHC/ByteOrder.hs
@@ -17,6 +17,9 @@
 
 module GHC.ByteOrder where
 
+-- Required for WORDS_BIGENDIAN
+#include "ghcautoconf.h"
+
 -- | Byte ordering.
 data ByteOrder
     = BigEndian    -- ^ most-significant-byte occurs in lowest address.
Index: b/libraries/base/GHC/IO/Encoding/CodePage/API.hs
===================================================================
--- a/libraries/base/GHC/IO/Encoding/CodePage/API.hs
+++ b/libraries/base/GHC/IO/Encoding/CodePage/API.hs
@@ -7,6 +7,9 @@ module GHC.IO.Encoding.CodePage.API (
     mkCodePageEncoding
   ) where
 
+-- Required for WORDS_BIGENDIAN
+#include "ghcautoconf.h"
+
 import Foreign.C
 import Foreign.Ptr
 import Foreign.Marshal
